### PR TITLE
allow setting matrix include and exclude options

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
here's an example run: https://github.com/evgeni/gha-playground/actions/runs/6610216830 (from this PR: https://github.com/evgeni/gha-playground/pull/4/files)

File: .github/workflows/foreman_plugin.yml [hunk 1]
```diff
@@ -30,9 +40,16 @@ jobs:
...
+          EOF
+          cat >matrix_exclude.json <<EOF
+          ${{ inputs.matrix_exclude }}
+          EOF
           {
             echo 'matrix<<EOF'
-            curl --silent --show-error --fail https://raw.githubusercontent.com/theforeman/foreman/${{ inputs.foreman_version }}/.github/matrix.json
+            python3 -c 'import json; matrix=json.load(open("matrix.json")); matrix["include"]=json.load(open("matrix_include.json")); matrix["exclude"]=json.load(open("matrix_exclude.json")); print(json.dumps(matrix));'
             echo 'EOF'
           } >> "$GITHUB_OUTPUT"
```
Comment (ekohl) [inline .github/workflows/foreman_plugin.yml:54]: I was debating whether it makes sense to completely do this in Python now.

**Review by ekohl — APPROVED** [review]
I'll merge this now, but it still feels a bit hacky.

**Comment by evgeni** [comment]
> I'll merge this now, but it still feels a bit hacky.

it ain't no beauty, agreed. but given a workflow has no way to have "files" unless you do an actual checkout (or are a "real" action), this was the best thing I could pull of quickly.

**Pull request merged**